EventListener/UserContextSubscriber.php

```diff
@@ -123,22 +130,24 @@ public function onKernelResponse(FilterResponseEvent $event)
             return;
         }
 
-        $response = $event->getResponse();
-        $vary = $response->getVary();
+        if ($this->addVaryOnHash) {
+            $response = $event->getResponse();
+            $vary = $response->getVary();
 
-        if ($event->getRequest()->headers->has($this->hashHeader)) {
-            if (!in_array($this->hashHeader, $vary)) {
-                $vary[] = $this->hashHeader;
-            }
-        } else {
-            foreach ($this->userIdentifierHeaders as $header) {
-                if (!in_array($header, $vary)) {
-                    $vary[] = $header;
+            if ($event->getRequest()->headers->has($this->hashHeader)) {
+                if (!in_array($this->hashHeader, $vary)) {
+                    $vary[] = $this->hashHeader;
+                }
+            } else {
```

Comment:
can you add a comment here that says something like 
```
/*
 * Additional safety: If for some reason we get requests without a user hash, vary on the credentials to avoid the caching proxy mixing up caches between users. For the hash lookup request, the headers are already added in onKernelRequest.
 */
```